File: typescript/src/semanticFunctions/semanticFunctionConfig.ts
// Copyright (c) Microsoft Corporation. All rights reserved.

import { IPromptTemplate } from './IPromptTemplate';
import { IPromptTemplateConfig } from './PromptTemplateConfig';

/**
 * Semantic function configuration.
 */
export class SemanticFunctionConfig {
    // Prompt template configuration.
    public readonly promptTemplateConfig: IPromptTemplateConfig;
    // Prompt template.
    public readonly promptTemplate: IPromptTemplate;

    /**
     * Constructor for SemanticFunctionConfig.
     * @param config Prompt template configuration.
     * @param template Prompt template.
     */
    public constructor(config: IPromptTemplateConfig, template: IPromptTemplate) {
        this.promptTemplateConfig = config;
        this.promptTemplate = template;
    }
}

File: java/semantickernel-api/src/main/ignore/java/com/microsoft/semantickernel/semanticfunctions/SemanticFunctionConfig.java
// Copyright (c) Microsoft. All rights reserved.
package com.microsoft.semantickernel.semanticfunctions;

/** Semantic function configuration */
public class SemanticFunctionConfig {
    private final PromptConfig config;

    /** Prompt template */
    private final PromptTemplate template;

    /**
     * Constructor for SemanticFunctionConfig.
     *
     * @param config Prompt template configuration.
     * @param template Prompt template.
     */
    public SemanticFunctionConfig(PromptConfig config, PromptTemplate template) {
        this.config = config;
        this.template = template;
    }

    public PromptConfig getConfig() {
        return config;
    }

    public PromptTemplate getTemplate() {
        return template;
    }
}
